scalafmt-tests/src/test/resources/default/Class.stat

```diff
@@ -551,3 +551,31 @@ object a {
   }
 
 }
+<<< early init 
+object CoreReflection extends {
+  val u: ru.type = ru
+  val mirror: u.Mirror = u.runtimeMirror(classOf[scala.meta.Tree].getClassLoader)
+} with scala.meta.internal.trees.Reflection with scala.meta.internal.tokens.Reflection
+>>>
+object CoreReflection extends {
+  val u: ru.type = ru
+  val mirror: u.Mirror =
+    u.runtimeMirror(classOf[scala.meta.Tree].getClassLoader)
+} with scala.meta.internal.trees.Reflection
+  with scala.meta.internal.tokens.Reflection
```

Comment:
Without this fix it's formatted as:
```scala
  object CoreReflection
      extends {
        val u: ru.type = ru
        val mirror: u.Mirror = u.runtimeMirror(classOf[scala.meta.Tree].getClassLoader)
      }
      with scala.meta.internal.trees.Reflection
      with scala.meta.internal.tokens.Reflection
```

Reply:
actually, i would think this new formatting is NOT desirable. why is `extends` treated differently than `with`?

how is this formatted:
```
object CoreReflection
      extends Foo(
        bar,
        baz
      )
      with scala.meta.internal.trees.Reflection
      with scala.meta.internal.tokens.Reflection
```

Reply:
From what I see, currently the logic is designed for the case of `extends-with` chain and then template body.
The early-init is a different case - there is a body between `extends` and `with` chain.

I think it would be better to support this construction if the required amount of changes is similar to what I proposed. That's the style how they were used + Scalafmt 2 worked worked well with it + migration to 3 triggers unwanted reformat.

Reply:
see #3094, it does something similar but consistent with configuration. the difference is indentation.